admin/zabbix/Makefile

```diff
@@ -74,68 +53,194 @@ define Package/zabbix/Default
   TITLE+= discovery/userparameters for network
-  DEPENDS = +zabbix-agentd +libubus-lua +lua
+  DEPENDS= +zabbix-agentd +libubus-lua +lua
 endef
 
 define Package/zabbix-extra-wifi
   $(call Package/zabbix/Default)
   TITLE+= discovery/userparameters for wifi
-  DEPENDS = +zabbix-agentd +libiwinfo-lua +libubus-lua +lua
+  DEPENDS= +zabbix-agentd +libiwinfo-lua +libubus-lua +lua
 endef
 
 define Package/zabbix-sender
   $(call Package/zabbix/Default)
   TITLE+= sender
+  PROVIDES:=zabbix-sender
+  VARIANT:=nossl
+  DEFAULT_VARIANT:=1
+endef
+
+define Package/zabbix-sender-openssl
+  $(call Package/zabbix/Default)
+  TITLE+= sender (with OpenSSL)
+  DEPENDS+= +libopenssl
+  PROVIDES:=zabbix-sender
+  VARIANT:=openssl
+endef
+
+define Package/zabbix-sender-gnutls
+  $(call Package/zabbix/Default)
+  TITLE+= sender (with GnuTLS)
+  DEPENDS+= +libgnutls
+  PROVIDES:=zabbix-sender
+  VARIANT:=gnutls
 endef
 
 define Package/zabbix-get
   $(call Package/zabbix/Default)
   TITLE+= get
+  PROVIDES:=zabbix-get
+  VARIANT:=nossl
+  DEFAULT_VARIANT:=1
 endef
 
-define Package/zabbix-server
+define Package/zabbix-get-openssl
+  $(call Package/zabbix/Default)
+  TITLE+= get (with OpenSSL)
+  DEPENDS+= +libopenssl
+  PROVIDES:=zabbix-get
+  VARIANT:=openssl
+endef
+
+define Package/zabbix-get-gnutls
+  $(call Package/zabbix/Default)
+  TITLE+= get (with GnuTLS)
+  DEPENDS+= +libgnutls
+  PROVIDES:=zabbix-get
+  VARIANT:=gnutls
+endef
+
+define Package/zabbix-server/Default
   $(call Package/zabbix/Default)
   TITLE+= server
-  DEPENDS += +ZABBIX_POSTGRESQL:libpq \
+  DEPENDS+= +ZABBIX_POSTGRESQL:libpq \
     +ZABBIX_MYSQL:libmariadbclient \
     +libevent2 \
     +fping
 endef
 
-define Package/zabbix-server-frontend
+define Package/zabbix-server
+  $(call Package/zabbix-server/Default)
+  PROVIDES:=zabbix-server
+  VARIANT:=nossl
+  DEFAULT_VARIANT:=1
+endef
+
+define Package/zabbix-server-openssl
+  $(call Package/zabbix-server/Default)
+  TITLE+= (with OpenSSL)
+  PROVIDES:=zabbix-server
+  DEPENDS+= +libopenssl
+  VARIANT:=openssl
+endef
+
+define Package/zabbix-server-gnutls
+  $(call Package/zabbix-server/Default)
+  TITLE+= (with GnuTLS)
+  PROVIDES:=zabbix-server
+  DEPENDS+= +libgnutls
+  VARIANT:=gnutls
+endef
+
+define Package/zabbix-server-frontend/Default
   $(call Package/zabbix/Default)
   TITLE+= server-frontend
-  DEPENDS += +php8 +php8-cgi +ZABBIX_POSTGRESQL:php8-mod-pgsql +ZABBIX_MYSQL:php8-mod-mysqli \
+  DEPENDS+= +php8 +php8-cgi +ZABBIX_POSTGRESQL:php8-mod-pgsql +ZABBIX_MYSQL:php8-mod-mysqli \
   +php8-mod-gd +php8-mod-bcmath +php8-mod-ctype +php8-mod-xmlreader +php8-mod-xmlwriter \
   +php8-mod-session +php8-mod-sockets +php8-mod-mbstring +php8-mod-gettext
 endef
 
-define Package/zabbix-proxy
+define Package/zabbix-server-frontend
+  $(call Package/zabbix-server-frontend/Default)
+  PROVIDES:=zabbix-server-frontend
+  VARIANT:=nossl
+  DEFAULT_VARIANT:=1
+endef
+
+define Package/zabbix-server-frontend-openssl
```

Comment:
The server frontend is a PHP application. And as I understand it, there is no difference between openssl and gnutls. If there is a variant needed, then the only difference is whether we use MySQL or PostgreSQL (but this is the next setup on update zabbix)